.circleci/config.yml

```diff
@@ -1,114 +1,89 @@
 
+jobs:
+  checkout:
+    executor: default
     steps:
       - checkout
-
-      - run:
-          name: Installing dependency for undercover gem
-          command: sudo apt-get install cmake
-
-      - <<: *install_bundler
-      - <<: *restore_bundle_cache
-      - <<: *bundle_install
-      - <<: *save_bundle_cache
-
-      - run:
-          name: Running overcommit
-          command: |
-            bundle exec overcommit -s
-            SKIP=AuthorEmail,AuthorName bundle exec overcommit -r
-
-  tests:
-    <<: *defaults
-
-    environment:
-      - CIRCLE_CI: true
-      - RAILS_ENV: test
-      - DB_USERNAME: root
-      - DB_PASSWORD: circle_test
-
+      - run: cp config/database.yml.ci config/database.yml
+      - persist_to_workspace:
+          root: ~/boilerplate-rails-api
+          paths:
+            - .
+  bundle:
+    executor: default
     steps:
-      - checkout
-
-      - run:
-          name: Installing dependency for undercover gem
-          command: sudo apt-get install cmake
-
-      - <<: *install_bundler
-      - <<: *restore_bundle_cache
-      - <<: *bundle_install
-      - <<: *save_bundle_cache
-
-      - run:
-          name: Preparing database config
-          command: cp config/database.yml.ci config/database.yml
-
-      - run:
-          name: Creating database
-          command: bundle exec rails db:create
-
-      - run:
-          name: Loading schema
-          command: bundle exec rails db:schema:load
-
-      - run:
-          name: Running tests
-          command: bundle exec rspec
-
-      - run:
-          name: Coverage
-          command: bundle exec undercover -c origin/develop
-
-      - store_artifacts:
-          path: ~/boilerplate-rails-api/coverage
-          destination: coverage
+      - attach_workspace:
+          at: ~/boilerplate-rails-api
+      - restore_cache:
+          keys:
+            - boilerplate-rails-api-{{ checksum "Gemfile.lock" }}
+            - boilerplate-rails-api-
+      - run: gem i bundler -v $(tail -1 Gemfile.lock | tr -d ' ')
+      - run: |
+          bundle check || bundle install --frozen --path vendor/bundle && bundle clean
+      - save_cache:
+          key: boilerplate-rails-api-{{ checksum "Gemfile.lock" }}
+          paths:
+            - vendor/bundle
+      - persist_to_workspace:
+          root: ~/boilerplate-rails-api
+          paths:
+            - vendor/bundle
+  overcommit:
+    executor: default
+    steps:
+      - attach_workspace:
+          at: ~/boilerplate-rails-api
+      - run: sudo apt-get install cmake
+      - run: |
+          echo 'export BUNDLER_VERSION=$(cat Gemfile.lock | tail -1 | tr -d " ")' >> $BASH_ENV
+          source $BASH_ENV
+          gem install bundler
+      - run: bundle install
+      - run: |
+          bundle exec overcommit -s
+          SKIP=AuthorEmail,AuthorName bundle exec overcommit -r
+  rspec:
+    executor: default
+    docker:
+      - image: circleci/ruby:2.6.5-node-browsers
```

Comment:
Why do we need `-browsers` here?

Reply:
We will use it when add active_admin and feature specs